pinot-segment-local/src/main/java/org/apache/pinot/segment/local/segment/creator/impl/stats/BigDecimalColumnPreIndexStatsCollector.java

```diff
@@ -113,8 +111,11 @@ public int getCardinality() {
 
   @Override
   public void seal() {
-    _sortedValues = _values.toArray(new BigDecimal[0]);
-    Arrays.sort(_sortedValues);
-    _sealed = true;
+    if (!_sealed) {
+      _sortedValues = _values.toArray(new BigDecimal[0]);
+      _values = null;
```

Comment:
Should we consider invoking `_values.clear()` method before setting it `null`? Same for all the other stats collectors.

Reply:
That shouldn't be needed. Once the `_values` is set to `null`, the whole map can be garbage collected. `_values.clear()` will introduce extra overhead, but not reduce the garbage